aiohttp/client_reqrep.py

```diff
@@ -86,23 +86,19 @@ def _gen_default_accept_encoding() -> str:
     return "gzip, deflate, br" if HAS_BROTLI else "gzip, deflate"
 
 
-@attr.s(auto_attribs=True, frozen=True, slots=True)
+@dataclasses.dataclass(frozen=True)
 class ContentDisposition:
     type: Optional[str]
     parameters: "MappingProxyType[str, str]"
     filename: Optional[str]
 
 
-@attr.s(auto_attribs=True, frozen=True, slots=True)
+@dataclasses.dataclass(frozen=True)
 class RequestInfo:
     url: URL
     method: str
     headers: "CIMultiDictProxy[str]"
-    real_url: URL = attr.ib()
-
-    @real_url.default
-    def real_url_default(self) -> URL:
-        return self.url
+    real_url: URL
```

Comment:
You can use `init=False` and write an `__init__` with the same behavior as it was with attrs.

Reply:
really not.
aiohttp always call `RequestInfo` with `real_url` argument; only tests use 3-arguments form.
Changing the signature is ok if there is no user code that instantiates `RequestInfo` explicitly.